de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/grid/GridViewPresenterImpl.java

```diff
@@ -248,9 +244,14 @@ public void onSuccess(ShareResourceLinkDialog dlg) {
         this.view.addDiskResourceSelectionChangedEventHandler(this);
     }
 
+    private void setupHandlers() {
+        clearHandlers();
```

Comment:
Yes! When using application wide event bus, we have to do this to avoid duplicate firing/handling of events.